File: flutter_app/lib/screens/login_screen.dart
import 'package:flutter/material.dart';
import '../theme/app_theme.dart';
import '../widgets/glass.dart';
import '../services/auth_service.dart';
import 'dashboard_shell.dart';

/// Responsive glassmorphism login — single column on phones, 60/40 split on
/// tablets/landscape. Auto-sizes to any device. Wired to Supabase Auth.
class LoginScreen extends StatefulWidget {
  const LoginScreen({super.key});

  @override
  State<LoginScreen> createState() => _LoginScreenState();
}

class _LoginScreenState extends State<LoginScreen> {
  final _email = TextEditingController();
  final _password = TextEditingController();
  final _auth = AuthService();
  bool _obscure = true;
  bool _loading = false;

  @override
  void dispose() {
    _email.dispose();
    _password.dispose();
    super.dispose();
  }

  Future<void> _submit() async {
    if (_email.text.trim().isEmpty || _password.text.isEmpty) {
      _toast('Enter your email and password.', error: true);
      return;
    }
    setState(() => _loading = true);
    try {
      await _auth.signIn(_email.text, _password.text);
      if (!mounted) return;
      Navigator.of(context).pushReplacement(
        MaterialPageRoute(builder: (_) => const DashboardShell()),
      );
    } catch (e) {
      _toast(_message(e), error: true);
    } finally {
      if (mounted) setState(() => _loading = false);
    }
  }

  String _message(Object e) {
    final s = e.toString();
    if (s.contains('Invalid login') || s.contains('invalid')) {
      return 'Invalid email or password.';
    }
    return s.replaceFirst('AuthException: ', '').replaceFirst('Exception: ', '');
  }

  void _toast(String msg, {bool error = false}) {
    ScaffoldMessenger.of(context).showSnackBar(SnackBar(
      content: Text(msg),
      backgroundColor: error ? AppColors.danger : AppColors.ink,
      behavior: SnackBarBehavior.floating,
    ));
  }

  @override
  Widget build(BuildContext context) {
    final wide = AppTheme.isWide(context);
    return Scaffold(
      body: GlassBackground(
        child: SafeArea(
          child: wide
              ? Row(children: [
                  Expanded(flex: 6, child: _brandPanel(context)),
                  Expanded(flex: 4, child: Center(child: _formArea(context))),
                ])
              : Center(
                  child: SingleChildScrollView(
                    padding: const EdgeInsets.symmetric(vertical: 24),
                    child: _formArea(context, showLogo: true),
                  ),
                ),
        ),
      ),
    );
  }

  Widget _formArea(BuildContext context, {bool showLogo = false}) {
    return ConstrainedBox(
      constraints: const BoxConstraints(maxWidth: 420),
      child: Padding(
        padding: const EdgeInsets.symmetric(horizontal: 20),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            if (showLogo) ...[
              _logoTile(64),
              const SizedBox(height: 14),
              Text('EZ-Workspace',
                  style: TextStyle(
                    fontSize: AppTheme.fluid(context, 22, 30),
                    fontWeight: FontWeight.w900,
                    letterSpacing: -0.5,
                    color: AppColors.ink,
                  )),
              const SizedBox(height: 4),
              Text('ENTERPRISE OPERATIONS PANEL',
                  style: TextStyle(
                    fontSize: 10, fontWeight: FontWeight.w700,
                    letterSpacing: 2.5, color: AppColors.muted,
                  )),
              const SizedBox(height: 22),
            ],
            GlassCard(
              padding: const EdgeInsets.all(24),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.stretch,
                children: [
                  Text('Internal Sign In',
                      style: TextStyle(
                        fontSize: 16, fontWeight: FontWeight.w900,
                        letterSpacing: 0.4, color: AppColors.ink,
                      )),
                  const SizedBox(height: 4),
                  Text('Access your operations dashboard',
                      style: TextStyle(fontSize: 12, color: AppColors.muted)),
                  const SizedBox(height: 22),
                  _label('CORPORATE EMAIL'),
                  const SizedBox(height: 6),
                  TextField(
                    controller: _email,
                    keyboardType: TextInputType.emailAddress,
                    autofillHints: const [AutofillHints.email],
                    decoration: glassField(hint: 'name@ezbillify.in', icon: Icons.mail_outline),
                  ),
                  const SizedBox(height: 16),
                  _label('SECURITY PASSWORD'),
                  const SizedBox(height: 6),
                  TextField(
                    controller: _password,
                    obscureText: _obscure,
                    onSubmitted: (_) => _submit(),
                    decoration: glassField(
                      hint: '••••••••',
                      icon: Icons.lock_outline,
                      suffix: IconButton(
                        icon: Icon(_obscure ? Icons.visibility_off_outlined : Icons.visibility_outlined,
                            size: 18, color: AppColors.muted),
                        onPressed: () => setState(() => _obscure = !_obscure),
                      ),
                    ),
                  ),
                  const SizedBox(height: 22),
                  _primaryButton(),
                  const SizedBox(height: 16),
                  Row(
                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                    children: [
                      Text('Forgot credentials?',
                          style: TextStyle(fontSize: 11, fontWeight: FontWeight.w700, color: AppColors.muted)),
                      _chip(Icons.verified_user_outlined, 'Secure'),
                    ],
                  ),
                ],
              ),
            ),
            const SizedBox(height: 18),
            Text('© ${DateTime.now().year} EZBillify Ventures Pvt Ltd · AGPL-3.0',
                textAlign: TextAlign.center,
                style: TextStyle(fontSize: 10, fontWeight: FontWeight.w500, color: AppColors.subtle)),
          ],
        ),
      ),
    );
  }

  Widget _primaryButton() {
    return Container(
      height: 52,
      decoration: BoxDecoration(
        color: AppColors.primary,
        borderRadius: BorderRadius.circular(AppRadius.md),
        boxShadow: [
          const BoxShadow(
            color: Colors.white,
            offset: Offset(-4, -4),
            blurRadius: 10,
          ),
          BoxShadow(
            color: AppColors.primary.withValues(alpha: 0.5),
            offset: const Offset(4, 4),
            blurRadius: 10,
          ),
        ],
      ),
      child: Material(
        color: Colors.transparent,
        child: InkWell(
          onTap: _loading ? null : _submit,
          borderRadius: BorderRadius.circular(AppRadius.md),
          child: Center(
            child: _loading
                ? const SizedBox(width: 22, height: 22, child: CircularProgressIndicator(strokeWidth: 2.4, color: Colors.white))
                : const Row(
                    mainAxisAlignment: MainAxisAlignment.center,
                    children: [
                      Text('AUTHORIZE ACCESS',
                          style: TextStyle(fontWeight: FontWeight.w900, letterSpacing: 1.5, fontSize: 12, color: Colors.white)),
                      SizedBox(width: 8),
                      Icon(Icons.arrow_forward, size: 16, color: Colors.white),
                    ],
                  ),
          ),
        ),
      ),
    );
  }

  Widget _label(String s) => Text(s,
      style: TextStyle(fontSize: 11, fontWeight: FontWeight.w800, letterSpacing: 0.8, color: AppColors.muted));

  Widget _chip(IconData icon, String s) => Container(
        padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
        decoration: BoxDecoration(
          color: AppColors.primary.withValues(alpha: 0.1),
          borderRadius: BorderRadius.circular(AppRadius.pill),
        ),
        child: Row(mainAxisSize: MainAxisSize.min, children: [
          Icon(icon, size: 13, color: AppColors.primary),
          const SizedBox(width: 5),
          Text(s, style: TextStyle(fontSize: 10, fontWeight: FontWeight.w800, color: AppColors.primary)),
        ]),
      );

  Widget _logoTile(double s) => Container(
        width: s, height: s,
        decoration: BoxDecoration(
          borderRadius: BorderRadius.circular(20),
          boxShadow: [BoxShadow(color: AppColors.primary.withValues(alpha: 0.35), blurRadius: 28, offset: const Offset(0, 12))],
        ),
        child: ClipRRect(
          borderRadius: BorderRadius.circular(20),
          child: Image.asset('assets/images/logo.png', fit: BoxFit.cover),
        ),
      );

  // Dark brand panel for tablets / landscape.
  Widget _brandPanel(BuildContext context) {
    Widget feature(IconData i, String t) => Padding(
          padding: const EdgeInsets.only(bottom: 14),
          child: Row(children: [
            Container(
              width: 38, height: 38,
              decoration: BoxDecoration(
                color: AppColors.primary.withValues(alpha: 0.16),
                borderRadius: BorderRadius.circular(10),
              ),
              child: Icon(i, size: 18, color: AppColors.primaryLight),
            ),
            const SizedBox(width: 12),
            Expanded(child: Text(t, style: TextStyle(fontSize: 14, color: Colors.white.withValues(alpha: 0.85)))),
          ]),
        );

    return Container(
      margin: const EdgeInsets.all(16),
      padding: const EdgeInsets.all(36),
      decoration: BoxDecoration(
        color: AppColors.ink,
        borderRadius: BorderRadius.circular(AppRadius.xl),
        boxShadow: [BoxShadow(color: AppColors.ink.withValues(alpha: 0.3), blurRadius: 40, offset: const Offset(0, 20))],
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Row(children: [
            _logoTile(44),
            const SizedBox(width: 12),
            const Text('EZ-Workspace',
                style: TextStyle(fontSize: 18, fontWeight: FontWeight.w900, color: Colors.white)),
          ]),
          Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Text('Run your entire company\nfrom one secure panel.',
                  style: TextStyle(
                    fontSize: AppTheme.fluid(context, 26, 40),
                    height: 1.05, fontWeight: FontWeight.w900, color: Colors.white,
                  )),
              const SizedBox(height: 24),
              feature(Icons.verified_user_outlined, 'Bank-grade security & full audit trails'),
              feature(Icons.apartment_outlined, 'HR, finance & operations unified'),
              feature(Icons.insights_outlined, 'Real-time insights across every team'),
            ],
          ),
          Row(children: [
            Icon(Icons.shield_outlined, size: 14, color: AppColors.primaryLight),
            const SizedBox(width: 6),
            Text('Encrypted · SSO-ready · Audit-logged',
                style: TextStyle(fontSize: 11, color: Colors.white.withValues(alpha: 0.45))),
          ]),
        ],
      ),
    );
  }
}

File: android_native/app/src/main/java/com/ezbillify/ezworkspace/ui/screens/LoginScreen.kt
package com.ezbillify.ezworkspace.ui.screens

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.BoxWithConstraints
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.widthIn
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowForward
import androidx.compose.material.icons.filled.Visibility
import androidx.compose.material.icons.filled.VisibilityOff
import androidx.compose.material.icons.outlined.Apartment
import androidx.compose.material.icons.outlined.Insights
import androidx.compose.material.icons.outlined.Lock
import androidx.compose.material.icons.outlined.Mail
import androidx.compose.material.icons.outlined.Shield
import androidx.compose.material.icons.outlined.VerifiedUser
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.OutlinedTextFieldDefaults
import androidx.compose.material3.SnackbarHost
import androidx.compose.material3.SnackbarHostState
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.text.input.VisualTransformation
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.ezbillify.ezworkspace.R
import com.ezbillify.ezworkspace.services.AuthService
import com.ezbillify.ezworkspace.theme.AppColors
import com.ezbillify.ezworkspace.theme.InterFontFamily
import com.ezbillify.ezworkspace.theme.neumorphicRaised
import com.ezbillify.ezworkspace.ui.components.NeuCard
import kotlinx.coroutines.launch

/** Responsive neumorphic login — single column on phones, 60/40 split on tablets/landscape. Wired to Supabase Auth. */
@Composable
fun LoginScreen(onSignedIn: () -> Unit) {
    val auth = remember { AuthService() }
    val scope = rememberCoroutineScope()
    val snackbar = remember { SnackbarHostState() }

    var email by remember { mutableStateOf("") }
    var password by remember { mutableStateOf("") }
    var obscure by remember { mutableStateOf(true) }
    var loading by remember { mutableStateOf(false) }

    fun submit() {
        if (email.isBlank() || password.isBlank()) {
            scope.launch { snackbar.showSnackbar("Enter your email and password.") }
            return
        }
        loading = true
        scope.launch {
            try {
                auth.signIn(email, password)
                onSignedIn()
            } catch (e: Exception) {
                snackbar.showSnackbar(e.message ?: "Invalid email or password.")
            } finally {
                loading = false
            }
        }
    }

    Box(Modifier.fillMaxSize().background(AppColors.scaffold)) {
        BoxWithConstraints(Modifier.fillMaxSize()) {
            val wide = maxWidth >= 820.dp
            if (wide) {
                Row(Modifier.fillMaxSize()) {
                    Box(Modifier.weight(6f)) { BrandPanel() }
                    Box(Modifier.weight(4f), contentAlignment = Alignment.Center) {
                        FormArea(email, { email = it }, password, { password = it }, obscure, { obscure = !obscure }, loading, ::submit, showLogo = false)
                    }
                }
            } else {
                Box(Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
                    Column(
                        Modifier.verticalScroll(rememberScrollState()).padding(vertical = 24.dp),
                        horizontalAlignment = Alignment.CenterHorizontally,
                    ) {
                        FormArea(email, { email = it }, password, { password = it }, obscure, { obscure = !obscure }, loading, ::submit, showLogo = true)
                    }
                }
            }
        }
        SnackbarHost(hostState = snackbar, modifier = Modifier.align(Alignment.BottomCenter).padding(16.dp))
    }
}

@Composable
private fun FormArea(
    email: String, onEmail: (String) -> Unit,
    password: String, onPassword: (String) -> Unit,
    obscure: Boolean, onToggleObscure: () -> Unit,
    loading: Boolean, onSubmit: () -> Unit,
    showLogo: Boolean,
) {
    Column(
        Modifier.widthIn(max = 420.dp).padding(horizontal = 20.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        if (showLogo) {
            LogoTile(64.dp)
            Spacer(Modifier.size(14.dp))
            Text("EZ-Workspace", fontFamily = InterFontFamily, fontWeight = FontWeight.Black, fontSize = 26.sp, color = AppColors.ink)
            Spacer(Modifier.size(4.dp))
            Text("ENTERPRISE OPERATIONS PANEL", fontFamily = InterFontFamily, fontWeight = FontWeight.Bold, fontSize = 10.sp, color = AppColors.muted)
            Spacer(Modifier.size(22.dp))
        }

        NeuCard(modifier = Modifier.fillMaxWidth()) {
            Column {
                Text("Internal Sign In", fontFamily = InterFontFamily, fontWeight = FontWeight.Black, fontSize = 16.sp, color = AppColors.ink)
                Spacer(Modifier.size(4.dp))
                Text("Access your operations dashboard", fontFamily = InterFontFamily, fontSize = 12.sp, color = AppColors.muted)
                Spacer(Modifier.size(22.dp))

                FieldLabel("CORPORATE EMAIL")
                Spacer(Modifier.size(6.dp))
                NeuTextField(
                    value = email, onValueChange = onEmail,
                    placeholder = "name@ezbillify.in",
                    leading = Icons.Outlined.Mail,
                    keyboardType = KeyboardType.Email,
                )
                Spacer(Modifier.size(16.dp))

                FieldLabel("SECURITY PASSWORD")
                Spacer(Modifier.size(6.dp))
                NeuTextField(
                    value = password, onValueChange = onPassword,
                    placeholder = "••••••••",
                    leading = Icons.Outlined.Lock,
                    isPassword = true,
                    obscured = obscure,
                    onToggleObscure = onToggleObscure,
                )
                Spacer(Modifier.size(22.dp))

                PrimaryButton(loading = loading, onClick = onSubmit)
                Spacer(Modifier.size(16.dp))

                Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                    Text("Forgot credentials?", fontFamily = InterFontFamily, fontWeight = FontWeight.Bold, fontSize = 11.sp, color = AppColors.muted)
                    Chip(Icons.Outlined.VerifiedUser, "Secure")
                }
            }
        }
        Spacer(Modifier.size(18.dp))
        Text(
            "© ${java.time.Year.now().value} EZBillify Ventures Pvt Ltd · AGPL-3.0",
            fontFamily = InterFontFamily, fontWeight = FontWeight.Medium, fontSize = 10.sp,
            color = AppColors.subtle,
        )
    }
}

@Composable
private fun FieldLabel(text: String) = Text(
    text, fontFamily = InterFontFamily, fontWeight = FontWeight.ExtraBold, fontSize = 11.sp, color = AppColors.muted,
)

@Composable
private fun NeuTextField(
    value: String,
    onValueChange: (String) -> Unit,
    placeholder: String,
    leading: androidx.compose.ui.graphics.vector.ImageVector,
    keyboardType: KeyboardType = KeyboardType.Text,
    isPassword: Boolean = false,
    obscured: Boolean = false,
    onToggleObscure: (() -> Unit)? = null,
) {
    OutlinedTextField(
        value = value,
        onValueChange = onValueChange,
        placeholder = { Text(placeholder, fontFamily = InterFontFamily, color = AppColors.ink.copy(alpha = 0.4f)) },
        leadingIcon = { Icon(leading, contentDescription = null, tint = AppColors.muted) },
        trailingIcon = if (isPassword) {
            {
                IconButton(onClick = { onToggleObscure?.invoke() }) {
                    Icon(
                        if (obscured) Icons.Filled.VisibilityOff else Icons.Filled.Visibility,
                        contentDescription = null, tint = AppColors.muted,
                    )
                }
            }
        } else null,
        visualTransformation = if (isPassword && obscured) PasswordVisualTransformation() else VisualTransformation.None,
        keyboardOptions = KeyboardOptions(keyboardType = keyboardType),
        singleLine = true,
        shape = RoundedCornerShape(14.dp),
        colors = OutlinedTextFieldDefaults.colors(
            focusedContainerColor = AppColors.inputFill,
            unfocusedContainerColor = AppColors.inputFill,
            focusedBorderColor = AppColors.primary,
            unfocusedBorderColor = androidx.compose.ui.graphics.Color.White.copy(alpha = 0.8f),
        ),
        modifier = Modifier.fillMaxWidth(),
    )
}

@Composable
private fun PrimaryButton(loading: Boolean, onClick: () -> Unit) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .height(52.dp)
            .clip(RoundedCornerShape(14.dp))
            .background(AppColors.primary)
            .clickable(enabled = !loading, onClick = onClick),
        contentAlignment = Alignment.Center,
    ) {
        if (loading) {
            CircularProgressIndicator(color = androidx.compose.ui.graphics.Color.White, strokeWidth = 2.4.dp, modifier = Modifier.size(22.dp))
        } else {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(
                    "AUTHORIZE ACCESS", fontFamily = InterFontFamily, fontWeight = FontWeight.Black,
                    fontSize = 12.sp, color = androidx.compose.ui.graphics.Color.White,
                )
                Spacer(Modifier.size(8.dp))
                Icon(Icons.Filled.ArrowForward, contentDescription = null, tint = androidx.compose.ui.graphics.Color.White, modifier = Modifier.size(16.dp))
            }
        }
    }
}

@Composable
private fun Chip(icon: androidx.compose.ui.graphics.vector.ImageVector, text: String) {
    Row(
        modifier = Modifier
            .clip(RoundedCornerShape(999.dp))
            .background(AppColors.primary.copy(alpha = 0.1f))
            .padding(horizontal = 12.dp, vertical = 6.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Icon(icon, contentDescription = null, tint = AppColors.primary, modifier = Modifier.size(13.dp))
        Spacer(Modifier.size(5.dp))
        Text(text, fontFamily = InterFontFamily, fontWeight = FontWeight.Black, fontSize = 10.sp, color = AppColors.primary)
    }
}

@Composable
private fun LogoTile(size: androidx.compose.ui.unit.Dp) {
    Box(
        modifier = Modifier.size(size).neumorphicRaised(cornerRadius = 20.dp),
    ) {
        Image(
            painter = painterResource(R.drawable.logo),
            contentDescription = "EZ-Workspace",
            contentScale = ContentScale.Crop,
            modifier = Modifier.fillMaxSize().clip(RoundedCornerShape(20.dp)),
        )
    }
}

@Composable
private fun BrandPanel() {
    Box(
        modifier = Modifier
            .fillMaxSize()
            .padding(16.dp)
            .clip(RoundedCornerShape(28.dp))
            .background(AppColors.ink)
            .padding(36.dp),
    ) {
        Column(Modifier.fillMaxSize(), verticalArrangement = Arrangement.SpaceBetween) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                LogoTile(44.dp)
                Spacer(Modifier.size(12.dp))
                Text("EZ-Workspace", fontFamily = InterFontFamily, fontWeight = FontWeight.Black, fontSize = 18.sp, color = androidx.compose.ui.graphics.Color.White)
            }
            Column {
                Text(
                    "Run your entire company\nfrom one secure panel.",
                    fontFamily = InterFontFamily, fontWeight = FontWeight.Black, fontSize = 34.sp, lineHeight = 38.sp,
                    color = androidx.compose.ui.graphics.Color.White,
                )
                Spacer(Modifier.size(24.dp))
                Feature(Icons.Outlined.VerifiedUser, "Bank-grade security & full audit trails")
                Feature(Icons.Outlined.Apartment, "HR, finance & operations unified")
                Feature(Icons.Outlined.Insights, "Real-time insights across every team")
            }
            Row(verticalAlignment = Alignment.CenterVertically) {
                Icon(Icons.Outlined.Shield, contentDescription = null, tint = AppColors.primaryLight, modifier = Modifier.size(14.dp))
                Spacer(Modifier.size(6.dp))
                Text(
                    "Encrypted · SSO-ready · Audit-logged",
                    fontFamily = InterFontFamily, fontSize = 11.sp, color = androidx.compose.ui.graphics.Color.White.copy(alpha = 0.45f),
                )
            }
        }
    }
}

@Composable
private fun Feature(icon: androidx.compose.ui.graphics.vector.ImageVector, text: String) {
    Row(Modifier.padding(bottom = 14.dp), verticalAlignment = Alignment.CenterVertically) {
        Box(
            modifier = Modifier
                .size(38.dp)
                .clip(RoundedCornerShape(10.dp))
                .background(AppColors.primary.copy(alpha = 0.16f)),
            contentAlignment = Alignment.Center,
        ) {
            Icon(icon, contentDescription = null, tint = AppColors.primaryLight, modifier = Modifier.size(18.dp))
        }
        Spacer(Modifier.size(12.dp))
        Text(text, fontFamily = InterFontFamily, fontSize = 14.sp, color = androidx.compose.ui.graphics.Color.White.copy(alpha = 0.85f))
    }
}
